src/promptflow/promptflow/_sdk/entities/_trace.py

```diff
@@ -252,11 +256,31 @@ def _from_spans(spans: typing.List[Span]) -> typing.Optional["LineRun"]:
             else:
                 # eager flow/arbitrary script
                 main_line_run_data = _LineRunData._from_root_span(span)
```

Comment:
main_line_run_data should calculate based on runs?

Reply:
ideally, yes; but it's kind of complicated to implement that logic - need some times, and might introduce some unexpected bugs... so I prefer to make this PR a workaround, and fully refine this when we persist line run in sqlite